### PR TITLE
Fix various warnings

### PR DESCRIPTION
**Pull request by dutkalex**
**_Describe your changes here:_**
Various minor bug fixes. Only considered the main target.

**_All these boxes must be checked by the reviewers before merging the pull request:_**

As a reviewer please read through all the code lines and make sure that the code is fully understood, bug free, well-documented and well-structured.

#### General
- [ ] The reviewer executed the new code features at least once and checked the results manually

- [ ] The code follows the [t8code coding guidelines](https://github.com/holke/t8code/wiki/Coding-Guideline)
- [ ] New source/header files are properly added to the Makefiles
- [ ] The code is well documented
- [ ] All function declarations, structs/classes and their members have a proper doxygen documentation
- [ ] All new algorithms and data structures are sufficiently optimal in terms of memory and runtime (If this should be merged, but there is still potential for optimization, create a new issue)

#### Tests
- [ ] The code is covered in an existing or new test case using Google Test

#### Github action

- [ ] The code compiles without warning in debugging and release mode, with and without MPI (this should be executed automatically in a github action)
- [ ] All tests pass (in various configurations, this should be executed automatically in a github action)

  If the Pull request introduces code that is not covered by the github action (for example coupling with a new library):
  - [ ] Should this use case be added to the github action?
  - [ ] If not, does the specific use case compile and all tests pass (check manually)

#### Scripts and Wiki

- [ ] If a new directory with source-files is added, it must be covered by the `script/find_all_source_files.scp` to check the indentation of these files.
- [ ] If this PR introduces a new feature, it must be covered in an example/tutorial and a Wiki article.

#### Licence

- [ ] The author added a BSD statement to `doc/` (or already has one)

### CODE REVIEW

**Comment by dutkalex** [comment]
The main target now compiles without any warning with `-Wall` (at least with Clang 17). Do you want to go for `-Wextra` too?

**Comment by sandro-elsweijer** [comment]
> The main target now compiles without any warning with `-Wall` (at least with Clang 17). Do you want to go for `-Wextra` too?

Hey @dutkalex,
we are keeping our code warning free with `-Wall` and `-pedantic`. No need for `-Wextra`. Thank you for your contribution, I will take a look at it :)

File: src/t8_forest/t8_forest_vtk.cxx [hunk 1]
```diff
@@ -114,10 +114,10 @@ typedef int (*t8_forest_vtk_cell_data_kernel) (t8_forest_t forest, const t8_loci
 
 #define T8_FOREST_VTK_QUADRATIC_ELEMENT_MAX_CORNERS 20
 /** Lookup table for number of nodes for curved eclasses. */
-const int t8_curved_eclass_num_nodes[T8_ECLASS_COUNT] = { 1, 3, 8, 6, 20, 10, 15, 13 };
+[[maybe_unused]] const int t8_curved_eclass_num_nodes[T8_ECLASS_COUNT] = { 1, 3, 8, 6, 20, 10, 15, 13 };
```
Comment (sandro-elsweijer) [inline src/t8_forest/t8_forest_vtk.cxx:117]: Used here: https://github.com/DLR-AMR/t8code/blob/aa2b436a07bad4066b35be0742a036dd3975a3e5/src/t8_forest/t8_forest_vtk.cxx#L172

File: src/t8_forest/t8_forest_vtk.cxx [hunk 2]
```diff
@@ -114,10 +114,10 @@ typedef int (*t8_forest_vtk_cell_data_kernel) (t8_forest_t forest, const t8_loci
...
 
 /** Lookup table for vtk types of curved elements */
-const int t8_curved_eclass_vtk_type[T8_ECLASS_COUNT] = { 1, 21, 23, 22, 25, 24, 26, 27 };
+[[maybe_unused]] const int t8_curved_eclass_vtk_type[T8_ECLASS_COUNT] = { 1, 21, 23, 22, 25, 24, 26, 27 };
```
Comment (sandro-elsweijer) [inline src/t8_forest/t8_forest_vtk.cxx:120]: Used here https://github.com/DLR-AMR/t8code/blob/aa2b436a07bad4066b35be0742a036dd3975a3e5/src/t8_forest/t8_forest_vtk.cxx#L295

**Review by sandro-elsweijer — CHANGES_REQUESTED** [review]
I have only minor comments. Thank you for the input!
After indenting your files `./scripts/indent_all_files.scp` from your t8code directory, the indentation check should also run through.
You can also add this script to your hooks so that it prevents you from committing unindented files: https://github.com/DLR-AMR/t8code/wiki/Coding-Guideline#git-indentation-and-spell-checking-workflow

File: src/t8_forest/t8_forest_vtk.cxx (hunk 1 above)
Comment (dutkalex) [reply src/t8_forest/t8_forest_vtk.cxx:117]: Yeah I tried without vtk on my machine, which explains why I got an unused variable warning...

File: src/t8_forest/t8_forest_vtk.cxx (hunk 2 above)
Comment (dutkalex) [reply src/t8_forest/t8_forest_vtk.cxx:120]: Same as above

**Comment by dutkalex** [comment]
I can't run the indentation script on my machine, but aside from that I think we're good :)

**Comment by sandro-elsweijer** [comment]
Hey @dutkalex,
me and most of the others are on vacation. But I will take a look as soon as I come back :)

File: src/t8_forest/t8_forest_vtk.cxx (hunk 1 above)
Comment (sandro-elsweijer) [reply src/t8_forest/t8_forest_vtk.cxx:117]: Can you please remove those markers again?

File: src/t8_forest/t8_forest_vtk.cxx (hunk 2 above)
Comment (sandro-elsweijer) [reply src/t8_forest/t8_forest_vtk.cxx:120]: Can you please remove those markers again?

**Review by sandro-elsweijer — COMMENTED** [review]
@dutkalex After removing those markers and indenting the code we can merge it :)

File: src/t8_forest/t8_forest_vtk.cxx (hunk 2 above)
Comment (dutkalex) [reply src/t8_forest/t8_forest_vtk.cxx]: Done

File: src/t8_forest/t8_forest_vtk.cxx (hunk 1 above)
Comment (dutkalex) [reply src/t8_forest/t8_forest_vtk.cxx]: Done

**Comment by dutkalex** [comment]
> @dutkalex After removing those markers and indenting the code we can merge it :)

The markers have been removed. Maybe it would be a better solution to add a ``#ifdef T8_WITH_VTK`` guard instead to avoid the warnings when building without VTK, but maybe that is material for another PR later...

**Comment by sandro-elsweijer** [comment]
> > @dutkalex After removing those markers and indenting the code we can merge it :)
> 
> @sandro-elsweijer The markers have been removed. Maybe it would be a better solution to add a `#ifdef T8_WITH_VTK` guard instead to avoid the warnings when building without VTK, but maybe that is material for another PR later...

Thats a good remark, I will take a look at it.
But I am wondering why the CI did not fail with your markers, since it runs with vtk as well.

**Comment by sandro-elsweijer** [comment]
see https://github.com/DLR-AMR/t8code/pull/1222

**Comment by sandro-elsweijer** [comment]
@dutkalex Unfortunately, I cannot merge main into your branch to fix the merge conflicts. You have to do that as well.

**Comment by dutkalex** [comment]
Aside from the indentation thing, you should be able to merge now I believe @sandro-elsweijer

**Comment by sandro-elsweijer** [comment]
> Aside from the indentation thing, you should be able to merge now I believe @sandro-elsweijer

Yes, but the indentation check is also required for a merge. Do you have problems installing the right clang format version?

**Comment by dutkalex** [comment]
> > Aside from the indentation thing, you should be able to merge now I believe @sandro-elsweijer
> 
> Yes, but the indentation check is also required for a merge. Do you have problems installing the right clang format version?

No. I get the following error when I try to run the script:
```
$ ./scripts/indent_all_files.scp
find: -regextype: unknown primary or operator
```

**Comment by sandro-elsweijer** [comment]
> > > Aside from the indentation thing, you should be able to merge now I believe @sandro-elsweijer
> > 
> > 
> > Yes, but the indentation check is also required for a merge. Do you have problems installing the right clang format version?
> 
> No. I get the following error when I try to run the script:
> 
> ```
> $ ./scripts/indent_all_files.scp
> find: -regextype: unknown primary or operator
> ```

That's weird and we did not have this error yet. I merged your changes into a new branch of mine to be able to indent it. You can find these changes here: https://github.com/DLR-AMR/t8code/pull/1223
This will probably be merged on Monday. Have a nice weekend!

**Comment by dutkalex** [comment]
From what I could gather on the internet, this might be related to the fact that macOS ships with a find whose options slightly differ from the standard unix ones... I did not investigate further though. Sorry for the inconvenience 😕

**Pull request closed without merging**